rollup-common.js

```diff
@@ -40,54 +40,75 @@ const skipBundleOutput = {
-  '_$litDirective$',
-  // TODO Decide on public API
-  // https://github.com/Polymer/lit-html/issues/1261
-  '_value',
-  '_setValue',
-  'createTreeWalker',
-  // Note, used in platform-support and reserved so platform-support can work
-  // x-version.
-  '_handlesPrepareStyles',
-  // TODO(kschaaf) TBD: lit-ssr required "private" fields (can be in
-  // crossPackagePropertyMangles once lit-ssr uses the rollup config)
-  // lit-html: AttributePart (used by render-lit-html)
-  '_commitValue',
-  // lit-html: Directive (used by render-lit-html)
-  '_resolve',
-  // updating-element: UpdatingElement (used by lit-element-renderer)
-  '_attributeToProperty',
-  // hydrate-support: LitElement (added by hydrate-support)
-  '_needsHydration',
-];
+// Private properties which should never be mangled. They need to be long/obtuse
+// to avoid collisions since they are used to brand values in positions that
+// accept any value. We don't use a Symbol for these to support mixing and
+// matching values from different versions.
+const reservedProperties = ['_$litType$', '_$litDirective$'];
 
-// Any private properties which we share between different _packages_ are
-// hard-coded here because they must never change between versions. Mangled
-// names are uppercase letters, in case we ever might want to
-// use lowercase letters for short, public APIs.
-// Note, these are used for `platform-support`.
-const crossPackagePropertyMangles = {
-  // lit-html: Template
-  _createElement: 'A',
-  _element: 'B',
-  _options: 'C',
-  // lit-html: NodePart
-  _startNode: 'D',
-  _endNode: 'E',
-  _getTemplate: 'F',
-  // lit-html: TemplateInstance
-  _template: 'H',
-  // updating-element: UpdatingElement
-  _didUpdate: 'S',
-  _controllers: 'T',
+// Private properties which should be stable between versions but are used on
+// unambiguous objects and thus are safe to mangle. These include properties on
+// objects accessed between packages or objects used as values which may be
+// accessed between different versions of a given package.
+//
+// By convention, stable properties should be prefixed with `_$` in the code so
+// they are easily identifiable as properties requiring version stability and
+// thus special attention.
+//
+// Mangled names are uppercase letters, in case we ever might want to use
+// lowercase letters for short, public APIs. Keep this list in order by mangled
+// name to avoid accidental re-assignments. When adding a name, add to the end
+// and choose the next letter.
+//
+// ONCE A MANGLED NAME HAS BEEN ASSIGNED TO A PROPERTY, IT MUST NEVER BE USED
+// FOR A DIFFERENT PROPERTY IN SUBSEQUENT VERSIONS.
+const stableProperties = {
+  // lit-html: Template (used by platform-support)
+  _$createElement: 'A',
+  _$element: 'B',
+  _$options: 'C',
+  // lit-html: NodePart (used by platform-support)
+  _$startNode: 'D',
+  _$endNode: 'E',
+  _$getTemplate: 'F',
+  // lit-html: TemplateInstance (used by platform-support)
+  _$template: 'G',
```

Comment:
I'm surprised TemplateResult.strings and values isn't in here already. We need those to be stable across versions. Can you add them?

Reply:
They're not prefixed with `_`, so they don't get mangled, and putting them in this list wouldn't do anything. Do we want them to go ahead and follow the `_$` convention also?

Reply:
Yeah. Removing the `_` was a temporary workaround